modules/smartytechBidAdapter.js

```diff
@@ -80,20 +82,57 @@ export const spec = {
         }
       }
 
+      // Add user IDs if available
+      const userIds = deepAccess(validBidRequest, 'userIdAsEids');
+      if (userIds && isArray(userIds) && userIds.length > 0) {
+        oneRequest.userIds = userIds;
+      }
+
+      // Add GDPR consent if available
+      if (bidderRequest && bidderRequest.gdprConsent) {
+        oneRequest.gdprConsent = {
+          gdprApplies: bidderRequest.gdprConsent.gdprApplies,
```

Comment:
Consider filtering `gdprApplies` so it’s only sent when it’s a boolean. This avoids sending `undefined` values.